feat: use make for scripting unifying the KEDA tools (#564)
kedacore/http-add-on: docs/developing.md

Follow the [install instructions](./install.md) to check out how to install and get this add-on up and running.

### In the Root Directory

The Makefile located in the root directory has targets useful for the whole project.

> All commands are case sensitive.

- `make build`: Builds all the binaries for local testing.

> - `make test`: Tests the entire codebase

Reviewer: Unit, e2e or both?

Author: unit tests, e2e test command is `make e2e-test`, I forgot to add it

Reviewer: Let's be more explicit in these docs then
```suggestion
- `make test`: Run all unit tests
```

and document e2e-test

Author: done

- `make docker-build`: Builds all docker images

> - `make docker-publish`: Pushes all docker images, building them first

Reviewer: ```suggestion
- `make docker-publish`: Build and push all Docker images
```

> - `make publish-multiarch`: Pushes all docker images, building them first for `linux/arm64` and `linux/amd64`

Reviewer: ```suggestion
- `make publish-multiarch`: Build and push all docker for `linux/arm64` and `linux/amd64`
```

> - `make manifests`: Builds all the manifest files for Kubernetes, it's important to build after every change

Reviewer: ```suggestion
- `make manifests`: Generate all the manifest files for Kubernetes, it's important to build after every change
```

What building are we eluding to at the end? Code or these manifests? We should add this to the PR template

Author: I don't get what do you mean

Author: If you meant as a check in the template to ensure that the manifests are generated, I don't think it's necessary because I already added a CI check that validates it (and shows a message requesting the execution of `make manifests`)
